lib/elements/clone.py

```diff
@@ -67,98 +58,94 @@ def clone_fill_angle(self):
+                    # So, next best thing is to apply it to a unit vector and pull the angle from that.
+                    return degrees(t.apply_to_point((1, 0).angle))
+
+            source_angle = angle_for_transform(source_transform)
+            cloned_angle = angle_for_transform(cloned_transform)
+            angle = cloned_angle - source_angle
+
+        elements = self.clone_to_elements(cloned_node)
         for element in elements:
-            stitch_groups = element.to_stitch_groups(last_patch)
-            patches.extend(stitch_groups)
-
-        source_node.set('transform', old_transform)
-        source_node.set(INKSTITCH_ATTRIBS['angle'], old_angle)
-        return patches
+            # We manipulate the element's node directly here instead of using get/set param methods, because otherwise
+            # we run into issues due to those methods' use of caching not updating if the underlying param value is changed.
 
-    def _get_rotation(self, transform, source_node, source_path):
-        try:
-            rotation = transform.rotation_degrees()
-        except ValueError:
-            source_path = CubicSuperPath(source_path)[0]
-            clone_path = Path(source_node.get_path()).transform(source_node.composed_transform())
-            clone_path = CubicSuperPath(clone_path)[0]
+            if self.clone_fill_angle is None:  # Normally, rotate the cloned element's angle by the clone's rotation.
+                element_angle = float(element.node.get(INKSTITCH_ATTRIBS['angle'], 0)) - angle
+            else:  # If clone_fill_angle is specified, override the angle instead.
+                element_angle = self.clone_fill_angle
 
-            angle_source = atan2(source_path[1][1][1] - source_path[0][1][1], source_path[1][1][0] - source_path[0][1][0])
-            angle_clone = atan2(clone_path[1][1][1] - clone_path[0][1][1], clone_path[1][1][0] - clone_path[0][1][0])
-            angle_embroidery = radians(-float(source_node.get(INKSTITCH_ATTRIBS['angle'], 0)))
+            if self.flip_angle:
+                element_angle = -element_angle
 
-            diff = angle_source - angle_embroidery
-            rotation = degrees(diff + angle_clone)
+            element.node.set(INKSTITCH_ATTRIBS['angle'], element_angle)
 
-            if self.flip_angle:
-                rotation = -degrees(diff - angle_clone)
+            stitch_groups = element.to_stitch_groups(last_patch)
+            patches.extend(stitch_groups)
 
-        return -rotation
+        # Remove the "manually cloned" tree.
+        parent.remove(cloned_node)
 
-    def get_clone_style(self, style_name, node, default=None):
-        style = node.style[style_name] or default
-        return style
+        return patches
 
     def center(self, source_node):
         transform = get_node_transform(self.node.getparent())
         center = self.node.bounding_box(transform).center
         return center
 
-    @property
-    def shape(self):
```

Comment:
It may not seem obvious, but we still need the shape property otherwise we may see a NotImplementedError on some occasions (for example when the user tries to add commands)

Reply:
Understood, it's been restored.